### PR TITLE
Psr 116

### PR DESCRIPTION
**Pull request by narendrapaduchuri**

### CODE REVIEW

File: app/controllers/RemoveMemberDetailsController.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (westwater) [inline app/controllers/RemoveMemberDetailsController.scala:18]: second license comment got added by accident?

File: app/pages/RemoveMemberDetailsPage.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import models.SchemeId.Srn
+import play.api.libs.json.{Format, JsPath, Json}
+
+case class RemoveMemberDetailsPage(srn: Srn) extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ toString
+
+  override def toString: String = "removeMemberDetails"
+
+  //implicit val formats: Format[MemberDetails] = Json.format[MemberDetails]
```
Comment (westwater) [inline app/pages/RemoveMemberDetailsPage.scala:28]: comment left ove

File: app/controllers/RemoveMemberDetailsController.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            ),
+          removeMemberDetails => {
+            if (removeMemberDetails) {
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers.remove(RemoveMemberDetailsPage(srn)))
+                _ <- saveService.save(updatedAnswers)
+              } yield Redirect(navigator.nextPage(RemoveMemberDetailsPage(srn), mode, updatedAnswers))
+            } else {
+              Future
+                .successful(Redirect(navigator.nextPage(RemoveMemberDetailsPage(srn), mode, request.userAnswers)))
```
[the remark this answers is not in the document]
Comment (narendrapaduchuri) [reply app/controllers/RemoveMemberDetailsController.scala:94]: No, these are for redirection purpose.

[the remark this answers is not in the document]
Comment (westwater) [reply app/controllers/RemoveMemberDetailsController.scala:78]: my bad I commented on the wrong line - I meant
https://github.com/hmrc/pension-scheme-return-frontend/blob/9f8f29a4f6cf0197d69cb4c6ba4026627f920ccc/app/controllers/RemoveMemberDetailsController.scala#L73
you might need to add a cleanup method to also delete the associated pages

File: test/controllers/RemoveMemberDetailsControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import controllers.RemoveMemberDetailsController.{form, viewModel}
+import eu.timepit.refined.{refineMV, refineV}
+import forms.YesNoPageFormProvider
+import models.{CheckMode, NameDOB, NormalMode}
+import pages.{AccountingPeriodPage, MemberDetailsPage}
+import views.html.YesNoPageView
+
+import java.time.LocalDate
+
+class RemoveMemberDetailsControllerSpec extends ControllerBaseSpec {
```
Comment (kurtlogan) [inline test/controllers/RemoveMemberDetailsControllerSpec.scala:28]: You need to add tests here that assert that the value has been removed from the cache

**Review by kurtlogan — CHANGES_REQUESTED** [review]

File: app/controllers/RemoveMemberDetailsController.scala [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          formWithErrors =>
+            Future.successful(
+              withMemberDetails(srn, index)(
+                nameDOB => BadRequest(view(formWithErrors, viewModel(srn, index, nameDOB, mode)))
+              )
+            ),
+          removeMemberDetails => {
+            if (removeMemberDetails) {
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers.remove(RemoveMemberDetailsPage(srn)))
```
[the remark this answers is not in the document]
Comment (narendrapaduchuri) [reply app/controllers/RemoveMemberDetailsController.scala:73]: It seems you are looking at different version, I have changed the RemoveMemberDetailsPage.scala to MemberDetailsPage as per our discussion. I am not sure why still is showing like that. I will go through it again.

**Comment by narendrapaduchuri** [comment]
No longer there changes valid, Since conflict while git push.

A pull request will be requested for PSR-116

**Pull request closed without merging**